src/main/java/org/sagebionetworks/bridge/services/ParticipantVersionService.java

```diff
@@ -0,0 +1,168 @@
+        participantVersion.setDataGroups(account.getDataGroups());
+        participantVersion.setLanguages(account.getLanguages());
+        participantVersion.setSharingScope(account.getSharingScope());
+        participantVersion.setTimeZone(account.getClientTimeZone());
+
+        // Convert study memberships into a map.
+        Map<String, String> studyMembershipMap = new HashMap<>();
+        for (Enrollment enrollment : account.getActiveEnrollments()) {
+            studyMembershipMap.put(enrollment.getStudyId(), enrollment.getExternalId());
+        }
+        participantVersion.setStudyMemberships(studyMembershipMap);
+
+        return participantVersion;
+    }
+
+    // Helper method which creates a participant version. This method automatically increments the version number.
+    // Package-scoped for unit tests.
+    void createParticipantVersion(ParticipantVersion participantVersion) {
+        checkNotNull(participantVersion);
+        checkNotNull(participantVersion.getAppId());
+        checkNotNull(participantVersion.getHealthCode());
+
+        // Get the old version, so we increment the version number.
+        long now = DateUtils.getCurrentMillisFromEpoch();
+        Optional<ParticipantVersion> existingOpt = participantVersionDao.getLatestParticipantVersionForHealthCode(
+                participantVersion.getAppId(), participantVersion.getHealthCode());
+        if (existingOpt.isPresent()) {
+            // Shortcut: If the participant version is unchanged, return early so we don't create a duplicate version.
+            ParticipantVersion existing = existingOpt.get();
+            if (isIdenticalParticipantVersion(existing, participantVersion)) {
+                return;
+            }
+
+            participantVersion.setParticipantVersion(existing.getParticipantVersion() + 1);
+
+            // createdOn can't be changed.
+            participantVersion.setCreatedOn(existing.getCreatedOn());
+        } else {
+            // Initial version is 1.
+            participantVersion.setParticipantVersion(1);
+
+            // If createdOn is not specified, set it now.
+            if (participantVersion.getCreatedOn() == 0) {
+                participantVersion.setCreatedOn(now);
+            }
+        }
+
+        // Update modifiedOn.
+        participantVersion.setModifiedOn(now);
+
+        // Create.
+        participantVersionDao.createParticipantVersion(participantVersion);
+    }
+
+    // Compares non-key attributes for participant versions. Returns true if they are the same, false if they are
+    // different.
+    // Package-scoped for unit tests.
+    static boolean isIdenticalParticipantVersion(ParticipantVersion oldVersion, ParticipantVersion newVersion) {
+        Map<String, Object> oldAttrMap = getParticipantVersionAttributes(oldVersion);
+        Map<String, Object> newAttrMap = getParticipantVersionAttributes(newVersion);
+        return oldAttrMap.equals(newAttrMap);
+    }
```

Comment:
Even with EqualsVerifier this is easier?

Reply:
We want to ignore fields like modifiedOn and participantVersion, which violates the .equals() contract.